### PR TITLE
Allowing NK to vary

### PR DESCRIPTION
**Pull request by andrewram4287**
Altering the minimization function to allow the variability of NK to vary over cluster and rank. Needed to alter a few functions in order to rebuild and guess for NK values correctly. Altered shape of data to be 3 instead of 5 to alllow for the function of `comparingGMMjax_NK` and `maxloglik_ptnnp_NK` to work

### CODE REVIEW

File: gmm/tensor.py [hunk 1]
```diff
@@ -220,9 +241,14 @@ def callback(_, state):
         options=opts,
     )
     tq.close()
-
-    optNK, optCP, optPT = vector_to_cp_pt(opt.x, rank, meanShape)
+    
+    if nk_rearrange is False:
+        optNK, optCP, optPT = vector_to_cp_pt(opt.x, rank, meanShape)
```
Comment (aarmey) [inline gmm/tensor.py:246]: If the only difference in the two branches of the if statement is nk_rearrange, you can just make this one line: `optNK, optCP, optPT = vector_to_cp_pt(opt.x, rank, meanShape, nk_rearrange= nk_rearrange)`

File: gmm/figures/figure7.py [hunk 2]
```diff
@@ -29,26 +29,28 @@ def makeFigure():
...
-        for ii, key in enumerate(coords)
-    ]
-    maximizedFactors[2] = reorder_table(maximizedFactors[2])
-
-    for i in range(0, len(maximizedFactors)):
-        sns.heatmap(data=maximizedFactors[i], vmin=0, ax=ax[i + 2])
+    # cmpCol = [f"Fac. {i}" for i in np.arange(1, fac + 1)]
+    # rankCol = [f"Cmp. {i}" for i in np.arange(1, rank + 1)]
+    # clustArray = [f"Clust. {i}" for i in np.arange(1, clust + 1)]
+    # coords = {"Cluster": clustArray, "Factor": cmpCol, "Drug": drugXA.coords["Drug"]}
```
Comment (aarmey) [inline gmm/figures/figure7.py:45]: Why comment all this out?

File: gmm/figures/figure10.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+"""
+This creates Figure 10.
```
Comment (aarmey) [inline gmm/figures/figure10.py:2]: Can we put the purpose of each figure into the docstring? I can get the figure number from the filename.

**Review by aarmey — CHANGES_REQUESTED** [review]
A few readability suggestions.

File: gmm/figures/figure7.py (hunk 2 above)
Comment (andrewram4287) [reply gmm/figures/figure7.py]: Forgot to change it back, on it!

**Review by aarmey — APPROVED** [review]
Looks good to merge!

**Pull request merged**